ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/Committer.scala

```diff
@@ -81,18 +81,19 @@ private[committer] trait Committer[PartialResult] extends SubmissionExecutor {
       inputState: DamlStateMap,
   ): (DamlLogEntry, Map[DamlStateKey, DamlStateValue]) =
     runTimer.time { () =>
-      val commitContext = CommitContext(inputState, recordTime, participantId)
+      val commitContext = CommitContext(inputState, recordTime, participantId, preExecute = false)
       val logEntry = runSteps(commitContext, submission)
       logEntry -> commitContext.getOutputs.toMap
     }
 
   def runWithPreExecution(
+      recordTime: Option[Time.Timestamp],
```

Comment:
What does this represent? Pre-execution has no record time because the record time is established at post-execution, after the deferred time bounds checks and conflict detection.

Reply:
Your other comments are useful but I think this one gets to the heart of why I said "I could be totally wrong".

This makes sense; I can understand why we should be establishing the record time during post-execution, not pre-execution. I added the logic here because it's the last point at which we talk about `DamlLogEntry`, rather than `Raw.Value`. I can see now why that's unhelpful.

I think we could _probably_ do this correctly in post-execution by deserializing the log entry, adding the record time, and then re-serializing it, but that's a lot of work for something that's only useful in testing. So I'm going to propose I close this PR and go with the slightly less efficient pre-executing integrity checker instead.

Thanks for explaining. :smile:

Reply:
My pleasure! Yeah I wouldn't add compute for the sake of testing, especially not at post-execution that needs to be as fast as possible. That's actually the whole point of pre-execution, i.e. pre-validating in parallel (optimistically) so that writing to the ledger can not only be safe but also significantly faster.